File: native/android/com/codename1/htmlform/HTMLFormNativeImpl.java
package com.codename1.htmlform;

import android.os.Looper;
import com.codename1.impl.android.AndroidImplementation;
import com.codename1.impl.android.AndroidNativeUtil;

public class HTMLFormNativeImpl {
    public boolean isMainThread() {
        return AndroidImplementation.getInstance().isJSDispatchThread();
    }

    public void notifyDispatchQueue() {
        AndroidImplementation.getInstance().runOnJSDispatchThread(new Runnable() {
            public void run() {
                HTMLForm.runQueuedEvent();
            }
        });
    }

    public boolean isSupported() {
        return true;
    }

}

File: native/win/com/codename1/htmlform/HTMLFormNativeImpl.cs
namespace com.codename1.htmlform
{


    public class HTMLFormNativeImpl : IHTMLFormNativeImpl
    {
        public bool isMainThread()
        {
            //return impl.SilverlightImplementation.dispatcher.HasThreadAccess;
            return com.codename1.ui.Display.getInstance().isEdt();
        }

        public void notifyDispatchQueue()
        {
            //impl.SilverlightImplementation.dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
           // {
           //     HTMLForm.runQueuedEvent();
            //});
            com.codename1.ui.Display.getInstance().callSerially(new RunQueuedEventClass());

        }

        public bool isSupported()
        {
            return true;
        }



    }

    class RunQueuedEventClass : object, java.lang.Runnable
    {
        public void run()
        {
            HTMLForm.runQueuedEvent();
        }
    }
   

   
}
